### PR TITLE
refactor: required parameters

### PR DESCRIPTION
**Pull request by LucasHengelhaupt**
## PR Type

[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no API changes)
[ ] Documentation content changes

## What Is the Current Behavior?
All component input parameters are optional. 

## What Is the New Behavior?

Mandatory component input parameters are declared as required to ensure that all necessary data is provided, catching potential errors early.

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

[AB#92105](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/92105)

### CODE REVIEW

**Review by SGrueber — CHANGES_REQUESTED** [review]

File: src/app/pages/account-payment/account-payment/account-payment.component.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { User } from 'ish-core/models/user/user.model';
 })
 export class AccountPaymentComponent implements OnInit, OnChanges {
   @Input() paymentMethods: PaymentMethod[];
```
[the remark this answers is not in the document]
Comment (LucasHengelhaupt) [reply src/app/pages/account-payment/account-payment/account-payment.component.ts:23]: made required

File: src/app/shell/header/sub-category-navigation/sub-category-navigation.component.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ import { InjectSingle } from 'ish-core/utils/injection';
 })
 export class SubCategoryNavigationComponent implements OnInit {
   @Input() view = 'auto';
-  @Input() categoryUniqueId: string;
-  @Input() subCategoriesDepth: number;
+  @Input({ required: true }) categoryUniqueId: string;
```
[the remark this answers is not in the document]
Comment (LucasHengelhaupt) [reply src/app/shell/header/sub-category-navigation/sub-category-navigation.component.ts:19]: changed order

File: src/app/shared/forms/components/form-control-feedback/form-control-feedback.component.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ interface FormErrorMessages {
 })
 export class FormControlFeedbackComponent implements DoCheck {
   @Input() messages: FormErrorMessages = {};
-  @Input() control: AbstractControl;
+  @Input({ required: true }) control: AbstractControl;
```
[the remark this answers is not in the document]
Comment (LucasHengelhaupt) [reply src/app/shared/forms/components/form-control-feedback/form-control-feedback.component.ts:15]: changed order

File: src/app/pages/account-profile-email/account-profile-email/account-profile-email.component.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ import { SpecialValidators } from 'ish-shared/forms/validators/special-validator
 })
 export class AccountProfileEmailComponent implements OnInit {
   @Input() error: HttpError;
-  @Input() currentUser: User;
+  @Input({ required: true }) currentUser: User;
```
[the remark this answers is not in the document]
Comment (LucasHengelhaupt) [reply src/app/pages/account-profile-email/account-profile-email/account-profile-email.component.ts:22]: changed order

File: src/app/pages/account-profile-company/account-profile-company/account-profile-company.component.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ import { markAsDirtyRecursive } from 'ish-shared/forms/utils/form-utils';
 })
 export class AccountProfileCompanyComponent implements OnInit {
   @Input() error: HttpError;
-  @Input() currentCustomer: Customer;
+  @Input({ required: true }) currentCustomer: Customer;
```
[the remark this answers is not in the document]
Comment (LucasHengelhaupt) [reply src/app/pages/account-profile-company/account-profile-company/account-profile-company.component.ts:22]: changed order

**Review by SGrueber — APPROVED** [review]

**Review by SGrueber — APPROVED** [review]

**Pull request merged**